content/swiftui/concepts/animation/animation.md

```diff
@@ -0,0 +1,107 @@
+---
+Title: 'Animation'
+Description: 'In response to state changes, produce seamless visual updates.'
+Subjects:
+  - 'iOS'
+  - 'Mobile Development'
+Tags:
+  - 'SwiftUI'
+  - 'SwiftUI ViewModifiers'
+  - 'iOS'
+CatalogContent:
+  - 'learn-iOS-course'
+  - 'path/iOS-path'
+---
+
+**Animation** is the ability to add smooth view changes to your view by changing the state of the view. There is two types of animation in swiftUI: Implicit animation and Explicit animation both of them will be explained in the sections below. it's very important to add animation to the user experience of an app, because it makes the app more interactive with the user and enhance the user experience of the app. And also Animating custom views can be done by conforming the view to the `Animatable` protocol, and telling SwiftUI about the value you want to animate.
+
+## Implicit Animation with .animation(_:value:)
+
+To be able to add animation to your view, you need first to have three things in your code:
+
+- A variable that changes to activate the animation. it must be `@State` variable to be able to reflect the changes to the view. 
+- Then after that we need to use a `Ternary ? operator`.
+- Then add the animation to the view.
+
+  This steps are essential to build animation implicitly or explicitly to your view. In the next example Implicit annimation will be used :
+
+```Swift
+struct ImplicitAnimation: View {
+    @State var isanimation = false // 1- @State variable that changes to activate the animation.
+    
+    var body: some View {
+        Button {
+            isanimation.toggle()
+        } label: {
+            Text( isanimation ? "Hello, World!" : "press me") // 2- Ternary Operator used to change the view based on the value of the isanimation.
+                .foregroundColor(.white)
+                .padding(.all)
+                .background( isanimation ? Color.black : Color.red) // 2- Ternary Operator used to change the view based onn the value of the isanimation.
+                .animation(.default, value: isanimation) // 3- This is the Implicit animation. this gets activated when the isanimation changes.
+        }
+
+    }
+}
+  ```
+> **Note:** `.animation` is a view modifier, which means that views That have this view applied to will be the only views to be animated.
```

Comment:
[Proofreading] Mentioning the context in which `animation` is used would make the explanation clearer. 
```suggestion
> **Note:** `.animation` in this code snippet, is a view modifier, which means that views That have this view applied to will be the only views to be animated.
```

Reply:
this is a general rule not only about this code snip